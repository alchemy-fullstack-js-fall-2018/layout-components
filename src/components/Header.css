.header {
    align-items: center;
    display: flex;
    flex-wrap: wrap;
    margin-bottom: 2em;
    .nav {
      text-decoration: none;
      color: #598381;
      &:hover {
        color: #177E89;
      }
    }
}


@media screen and (min-width: 700px) {

  .header {
    align-items: center;
    display: flex;
    margin-bottom: 2em;
    #logo, .nav {
      flex: 1;
    }

    .nav {
      display: flex;
      justify-content: space-between;
      text-decoration: none;
      color: #598381;
      &:hover {
        color: #177E89;
      }
    }
  }
}
