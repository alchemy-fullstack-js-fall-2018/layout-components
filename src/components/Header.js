import Component from './Component';
import styles from './Header.css';

export default class Header extends Component {
  render() {
    const links = this.props.links.map(link => {
      return `<a class="${styles.nav}"href="/${link}">${link}</a>`;
    });

    return `
      <header class="${styles.header}">
        <h2 id="${styles.logo}">Mack BG<h2>
        <nav class="${styles.nav}">
          ${links.join(' ')}
        </nav>
      </header>`;
  }
}
