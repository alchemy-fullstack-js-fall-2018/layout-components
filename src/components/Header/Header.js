import Component from '../Component/Component';

import styles from './Header.css';
export default class Header extends Component {

  render() {

    return `
      <header class="${styles.header}">
        <h1>Football Teams</h1>
      </header>
    `;
  }


}
