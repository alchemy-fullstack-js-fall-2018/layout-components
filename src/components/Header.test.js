import Header from './Header';

describe('Header Component', () => {
  it('should be a header', () => {
    const header = new Header({
      links: ['About', 'Unnecessary Fun Facts', 'Dog Photos']
    });

    expect(header.render()).toMatchSnapshot();
  });
});
