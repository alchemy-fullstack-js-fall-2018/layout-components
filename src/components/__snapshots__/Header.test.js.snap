// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header Component should be a header 1`] = `
"
      <header class=\\"header\\">
        <h2 id=\\"logo\\">Mack BG<h2>
        <nav class=\\"nav\\">
          <a class=\\"nav\\"href=\\"/About\\">About</a> <a class=\\"nav\\"href=\\"/Unnecessary Fun Facts\\">Unnecessary Fun Facts</a> <a class=\\"nav\\"href=\\"/Dog Photos\\">Dog Photos</a>
        </nav>
      </header>"
`;
